### ISSUE TITLE
Extend OpenPGP signing support with Stateless OpenPGP

### ISSUE THREAD

**Issue by dvzrv** [systemd/mkosi#3042]
### mkosi commit the issue has been seen with

https://github.com/systemd/mkosi/commit/baab5c5014e9332c6c85d4408d7b08d7907b40ce

### Used host distribution

Arch Linux

### Used target distribution

Arch Linux

### Linux kernel version used

6.10.10-arch1-1

### CPU architectures issue was seen on

x86_64

### Unexpected behaviour you saw

Currently it is only possible to sign the SHA256SUMS file using `gpg`.
However, gnupg is a complex thing to get working properly (see #3040) and also [starts to diverge from OpenPGP compatibility in >=2.4](https://wiki.archlinux.org/title/GnuPG#OpenPGP_compatibility).

Better, more simple (and stateless) alternatives for the "sign an artifact" use-case exist with [Stateless OpenPGP](https://wiki.archlinux.org/title/Stateless_OpenPGP) (SOP). With [`rsop`](https://codeberg.org/heiko/rsop) we even have a SOP implementation with smartcard support!

To that end, it would be great to extend the signing capabilities in mkosi by allowing to set a specific OpenPGP implementation (e.g. using an `OpenPGPTool` / `--openpgp-tool` option in the `[Validation]` section).
The `Key` option would then need to support setting either an [OpenPGP fingerprint](https://openpgp.dev/book/glossary.html#term-OpenPGP-Fingerprint) (in the case of `gpg`) or a path to a key (or certificate) in the case of SOP implementations.

Additionally, it would be good to rename the currently used options in the `[Validation]` section: `Sign` to `OpenPGPSign` and `Key` to `OpenPGPKey` (that way one could use several signing schemes in parallel, e.g. the ones discussed in https://github.com/systemd/mkosi/issues/624).

Closing, I would like to add, that `.gpg` is not a good signature suffix for OpenPGP signatures (e.g. `.sig` might be better as it is not OpenPGP implementation specific and indicates that it is a signature and not possibly a certificate or a keyring, etc.), but I guess that is currently somewhat fixed due to how [sysupdate.d](https://man.archlinux.org/man/sysupdate.d.5.en) expects it. :smiling_face_with_tear: 

### Used mkosi config

```ini
[Output]
Format=disk
SplitArtifacts=yes

[Content]
Bootable=yes
Bootloader=systemd-boot
Hostname=arch
Packages=
	base
	linux
	nftables
	openssh
	systemd
UnifiedKernelImageFormat=%i-%v+&c
UnifiedKernelImages=yes

[Distribution]
Architecture=x86-64
Distribution=arch

[Host]
RuntimeScratch=no
RuntimeSize=12G

[Validation]
Checksum=yes
Key=<my-key-fingerprint>
Sign=yes
```

### mkosi output

```sh
‣  Signing SHA256SUMS…
gpg: using "991F6E3F0765CF6295888586139B09DA5BF0D338" as default secret key for signing
```

**Comment by q2dg** [comment]
What about minisign? (https://jedisct1.github.io/minisign)

**Comment by wiktor-k** [comment]
> What about minisign? (https://jedisct1.github.io/minisign)

I think this is a separate issue. SOP produces OpenPGP artifacts so systemd may verify it just fine with their existing gpg setup. Minisign would require changes on both ends.

I'm not saying it's a bad idea... just a little bit more complex.
